Movecraft/src/main/java/net/countercraft/movecraft/features/status/StatusManager.java

```diff
@@ -70,9 +71,13 @@ private StatusUpdateTask(@NotNull Craft craft) {
         @Override
         public @Nullable Effect get() {
             Counter<Material> materials = new Counter<>();
+            Counter<RequiredBlockEntry> flyblocks = new Counter<>();
+            Counter<RequiredBlockEntry> moveblocks = new Counter<>();
             int nonNegligibleBlocks = 0;
             int nonNegligibleSolidBlocks = 0;
             double fuel = 0;
+            final var flyblocksList = crafttype.getRequiredBlockProperty(CraftType.FLY_BLOCKS);
+            final var moveblocksList = crafttype.getRequiredBlockProperty(CraftType.MOVE_BLOCKS);
```

Comment:
These are only used once, they can be put directly into the for loops below.

Reply:
oh yeah before they were called too many times, that is why they were there now there is no need for them

Reply:
wait not it is called for each material in the counter class